lib/api/model/events.dart

```diff
@@ -900,16 +901,16 @@ class TypingEvent extends Event {
     required this.recipientIds,
     required this.streamId,
     required this.topic,
-  });
+  }) : assert(isSortedWithoutDuplicates(recipientIds ?? []));
 
   static Object? _readSenderId(Map<Object?, Object?> json, String key) {
     return (json['sender'] as Map<String, dynamic>)['user_id'];
   }
 
   static List<int>? _recipientIdsFromJson(Object? json) {
     if (json == null) return null;
-    return (json as List<Object?>).map(
-      (item) => (item as Map<String, Object?>)['user_id'] as int).toList();
+    return (json as Iterable<Object?>).map(
```

Comment:
nit:
```suggestion
    return (json as List<Object?>).map(
```

That keeps this code resembling the generated deserialization code when it comes to boring parts like casting pieces of the data to their expected types. There isn't some expected situation where this would be a non-list iterable, right?